haskell is getting a lot of

waitForProcess: failed ((ipc/mig) server type check failure)

Force putting save_data on the stack rather than in SSE register, to avoid
getting bogus RPC in case a haskell signal triggers some SSE clobber.

XXX: This is only temporary to fix the haskell issue, and is waiting for fixed
SSE state restoration.

Index: glibc-2.38/hurd/intr-msg.c
===================================================================
--- glibc-2.38.orig/hurd/intr-msg.c
+++ glibc-2.38/hurd/intr-msg.c
@@ -79,7 +79,7 @@ _hurd_intr_rpc_mach_msg (mach_msg_header
   mach_msg_bits_t msgh_bits;
   mach_port_t remote_port;
   mach_msg_id_t msgid;
-  struct clobber save_data;
+  volatile struct clobber save_data;
 
   if ((option & (MACH_SEND_MSG|MACH_RCV_MSG)) != (MACH_SEND_MSG|MACH_RCV_MSG)
       || _hurd_msgport_thread == MACH_PORT_NULL)
